tests/vmi_networking_test.go

```diff
@@ -402,4 +405,51 @@ var _ = Describe("Networking", func() {
 		})
 	})
 
+	checkMacAddress := func(vmi *v1.VirtualMachineInstance, expectedMacAddress string, prompt string) {
+		err := tests.CheckForTextExpecter(vmi, []expect.Batcher{
+			&expect.BSnd{S: "\n"},
+			&expect.BExp{R: prompt},
+			&expect.BSnd{S: "cat /sys/class/net/eth0/address\n"},
+			&expect.BExp{R: expectedMacAddress},
+		}, 15)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	Context("VirtualMachineInstance with custom MAC address", func() {
+		It("should configure custom MAC address", func() {
+			By("checking eth0 MAC address")
+			deadbeafVMI := tests.NewRandomVMIWithCustomMacAddress()
+			_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(deadbeafVMI)
+			Expect(err).ToNot(HaveOccurred())
+
+			waitUntilVMIReady(deadbeafVMI, tests.LoggedInAlpineExpecter)
+			checkMacAddress(deadbeafVMI, deadbeafVMI.Spec.Domain.Devices.Interfaces[0].MacAddress, "localhost:~#")
+		})
+	})
+
+	Context("VirtualMachineInstance with custom MAC address in non-conventional format", func() {
+		It("should configure custom MAC address", func() {
+			By("checking eth0 MAC address")
+			beafdeadVMI := tests.NewRandomVMIWithCustomMacAddress()
+			beafdeadVMI.Spec.Domain.Devices.Interfaces[0].MacAddress = "BE-AF-00-00-DE-AD"
+			_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(beafdeadVMI)
+			Expect(err).ToNot(HaveOccurred())
+
+			waitUntilVMIReady(beafdeadVMI, tests.LoggedInAlpineExpecter)
+			checkMacAddress(beafdeadVMI, "be:af:00:00:de:ad", "localhost:~#")
+		})
+	})
+
+	Context("VirtualMachineInstance with custom MAC address and slirp interface", func() {
```

Comment:
Isn't this test case already covered by the slirp functional tests?

Reply:
No, we don't check that correct MAC address is set inside the guest there.